label-external-pull-request/action.yml

```
@@ -0,0 +1,51 @@
+name: Label External Pull Requests
+description: Checks if a PR is a fork and applies an external label accordingly
+
+inputs:
+  github_token:
+    description: The GitHub token to use read member status
+    required: true
+  team:
+    description: The team slug for external collaborators
+    required: true
+  org:
+    description: The organization to check membership in
+    required: true
+  username:
+    description: The username to check membership for
+    required: true
+
+runs:
+  using: composite
+  steps:
+    - name: Check if fork and label
+      shell: bash
+      env:
+        GITHUB_TOKEN: ${{ github.token }}
+      run: |
+            if [ '${{ github.event.pull_request.head.repo.fork}}'=='false' ]; then 
+                gh pr edit ${{github.event.pull_request.html_url}} --add-label "external"; 
+            fi
+    
+    - uses: actions/github-script@v7
+      id: set-result
+      with: 
+        github-token: ${{ inputs.github_token }}
```

```suggestion
      with: 
        github-token: ${{ inputs.github_token }}
        result-encoding: string
```

By default, this action attempts to encode the return type as JSON. But if we need to return something outside of JSON, we have to tell it to return the output as raw string.  [Ref](https://github.com/actions/github-script/blob/3908079ba1e7bce10117ad701c321d07e89017a9/action.yml#L23-L25)